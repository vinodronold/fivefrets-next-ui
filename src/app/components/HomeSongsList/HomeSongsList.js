import React, { PureComponent } from 'react'
import { bool, arrayOf, object } from 'prop-types'
import SongList from '../SongList'

export default class HomeSongsList extends PureComponent {
  static propTypes = {
    isError: bool,
    songs: arrayOf(object)
  }

  static defaultProps = {
    isError: false,
    songs: []
  }

  componentWillMount() {
    if (!(this.props.songs.length > 0)) {
      this.props.FetchHomePageSongs()
    }
  }
  render() {
    const { isError, songs } = this.props
    return <SongList isError={isError} songs={songs} />
  }
}
