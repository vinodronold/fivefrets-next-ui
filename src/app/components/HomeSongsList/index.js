import HomeSongsList from './HomeSongsList'
export default HomeSongsList
