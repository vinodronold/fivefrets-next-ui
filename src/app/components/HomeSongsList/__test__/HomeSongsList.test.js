import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import { spy } from 'sinon'
import HomeSongsList from '../HomeSongsList'

jest.mock('next/router')

describe('components/HomeSongsList', () => {
  it('should match snapshot', () => {
    const _home_songs_list = renderer.create(<HomeSongsList FetchHomePageSongs={jest.fn()} />)
    expect(_home_songs_list).toMatchSnapshot()
  })

  it('FetchHomePageSongs should be called when Songs not availble', () => {
    const _spy_FetchHomePageSongs = spy()
    mount(<HomeSongsList FetchHomePageSongs={_spy_FetchHomePageSongs} />)
    expect(_spy_FetchHomePageSongs.called).toEqual(true)
  })

  it('FetchHomePageSongs should not be called when Songs availble', () => {
    const _spy_FetchHomePageSongs = spy()
    mount(<HomeSongsList FetchHomePageSongs={_spy_FetchHomePageSongs} songs={[{ songs: 'test' }]} />)
    expect(_spy_FetchHomePageSongs.notCalled).toEqual(true)
  })
})
