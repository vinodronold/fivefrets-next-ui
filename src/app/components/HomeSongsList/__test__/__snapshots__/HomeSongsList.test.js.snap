// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/HomeSongsList should match snapshot 1`] = `
<div
  className="jsx-1057316989 container"
>
  <div
    className="jsx-1057316989 loader"
  >
    <div
      className="jsx-2763792101 jsx-3891026477 bar"
    />
    <div
      className="jsx-2763792101 jsx-1900015026 bar"
    />
    <div
      className="jsx-2763792101 jsx-2551557105 bar"
    />
    <div
      className="jsx-2763792101 jsx-2892672692 bar"
    />
    <div
      className="jsx-2763792101 jsx-216973047 bar"
    />
  </div>
  <div
    className="jsx-4247411757 jsx-3754337244"
    style={Object {}}
  >
    LOADING...
  </div>
</div>
`;
